### PR TITLE
JS: support more regular expressions in js/incomplete-multi-character-sanitization

### PR DESCRIPTION
**Pull request by erik-krogh**
CVE-2021-29489: TP/TN

The PR gains [this result in the existing test-case](https://github.com/github/codeql/pull/6462/files#diff-ad110d7f907a8210a9a4feab0799ceb8f5bda7d3666adc39c408e5b6ad9a8bbaR137). 
I'm not sure why, and I'm also not sure why that was marked as `OK` in the first place. 

[An evaluation shows some new TPs](https://github.com/dsp-testing/erik-krogh-dca/blob/run/repeat-eval-confirmation-run/reports/alert-comparison.md). (There are no lost results, the alert message just changed). 
And [a performance evaluation looks OK](https://github.com/dsp-testing/erik-krogh-dca/tree/run/repeat-evaluation-with-more-caches/reports).   

([Python performance also looks good](https://github.com/dsp-testing/erik-krogh-dca/tree/run/pr-6462-python-performance-check/reports)).

### CODE REVIEW

**Review by yoff — DISMISSED** [review]
Looks good from the Python side 👍

File: javascript/ql/src/Security/CWE-116/IncompleteMultiCharacterSanitization.ql [hunk 1]
```diff
@@ -56,31 +56,66 @@ DangerousPrefix getADangerousMatchedPrefix(EmptyReplaceRegExpTerm t) {
...
+pragma[noinline]
+DangerousPrefixSubstring getADangerousMatchedChar(EmptyReplaceRegExpTerm t) {
+  t.isNullable() and result = ""
+  or
+  t.getAMatchedString() = result
+  or
+  exists(ReDoSUtil::CharacterClass cc |
+    cc = ReDoSUtil::getCanonicalCharClass(t) and
+    cc.matches(result) and
+    not cc.matches(">")
```
Comment (esbena) [inline javascript/ql/src/Security/CWE-116/IncompleteMultiCharacterSanitization.ql:73]: I have tried to add a suggestion here with a comment that explains the reason for excluding `>` explicitly, but I can't do much better than `// needed to exclude /<[^<]*>/`, which seems a bit too ad hoc. Could we move the blacklist elsewhere, or formulate the general principle it implements.

Comment (erik-krogh) [reply javascript/ql/src/Security/CWE-116/IncompleteMultiCharacterSanitization.ql:73]: I think it's OK to just add a comment explaining why it's there. 
That comment would also explain that the `CharacterClass` matching is only relevant for the HTML-tags regexps. 
I'll try to get something in there. 

I don't think we can make the code much prettier without basically duplicating a bunch of the implementation.

Comment (esbena) [reply javascript/ql/src/Security/CWE-116/IncompleteMultiCharacterSanitization.ql:73]: Ok. Let's keep it simple then. 👍🏻

**Review by esbena — APPROVED** [review]

**Pull request merged**